packages/client/src/clients/in-app-messages/message-client.ts

```diff
@@ -69,75 +73,89 @@ export class InAppMessagesClient {
       return;
     }
 
-    // TODO: Move to method on channel client
     // Set the loading type based on the request type it is
-    this.channelClient.store.setState((state) => ({
-      ...state,
-      queries: {
-        ...state.queries,
-        [this.queryKey]: {
-          ...queryState,
-          networkStatus: options.__loadingType ?? NetworkStatus.loading,
-          loading: true,
-        },
-      },
-    }));
-
-    // TODO: Move to method on user.getInAppMessages
-    const result = await this.knock.client().makeRequest({
-      method: "GET",
-      url: `/v1/users/${this.knock.userId}/in-app-messages/${this.channelClient.channelId}/${this.messageType}`,
-      params: queryParams,
+    this.channelClient.setQueryState(this.queryKey, {
+      ...queryState,
+      networkStatus: options.__loadingType ?? NetworkStatus.loading,
+      loading: true,
     });
 
-    if (result.statusCode === "error" || !result.body) {
-      this.channelClient.store.setState((state) => ({
-        ...state,
-        queries: {
-          ...state.queries,
-          [this.queryKey]: {
-            ...queryState,
-            networkStatus: NetworkStatus.error,
-            loading: false,
-          },
-        },
-      }));
+    try {
+      const response = await this.knock.user.getInAppMessages<TContent, TData>({
+        channelId: this.channelClient.channelId,
+        messageType: this.messageType,
+        params,
+      });
 
-      return {
-        status: result.statusCode,
-        data: result.error || result.body,
+      const queryInfo: InAppMessagesQueryInfo = {
+        loading: false,
+        networkStatus: NetworkStatus.ready,
+        data: {
+          messageIds: response.entries.map((iam) => iam.id),
+          pageInfo: response.pageInfo,
+        },
       };
-    }
 
-    const response: InAppMessageResponse = {
-      entries: result.body.entries,
-      pageInfo: result.body.page_info,
-    };
+      this.channelClient.store.setState((state) => {
```

Comment:
And I think this can be also abstracted to a helper called `setQueryResult` or `setQueryResp` that:
* Takes the query key and the response
* Builds a query info and updates both `messages` and `queries`

So you end up with a set of two helpers: `setQueryStatus` and `setQueryResult`, of which the latter is used when we have a successful response. Wdyt?

Reply:
Yeah, I'm good with separating this into two more specific helpers